plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImagePage.java

```diff
@@ -172,6 +173,11 @@ protected IStatus run(IProgressMonitor monitor) {
 			OpenShiftUIActivator.getDefault().getLogger().logError(message, e);
 		}
 	}
+	
+	@Override
+	protected void onPageActivated(DataBindingContext dbc) {
+		loadResources(dbc);
```

Comment:
Can it be an issue, that when deploy image page is not the first one (for instance, wizard is started from the docker explorer, and the connection page is before it), going back and forward will always cause resource loading, even if connection was not changed (though, this use case is very minor).